packages/omgidl-serialization/src/MessageReader.ts

```diff
@@ -67,9 +67,12 @@ export class MessageReader<T = unknown> {
         ? this.readStructType(deserInfo, reader, options)
         : this.readUnionType(deserInfo, reader, options);
 
-    if (readMemberHeader) {
+    if (readMemberHeader && this.#unusedEmHeader?.readSentinelHeader !== true) {
       reader.sentinelHeader();
     }
+    // clear emHeader for aggregated type, since if it was defined, it would've likely been used
+    // as the sentinel header
+    this.#unusedEmHeader = undefined;
```

Comment:
not totally sure I understand why this is needed – so in lieu of understanding it I will ask: will one of your tests fail if it is removed?

Reply:
Yes. If it is removed an inner struct's sentinel header(representing the end of a struct), if encountered instead of an emHeader due to an absent final field(s) within the inner struct, will not be cleared, causing subsequent fields read in the outer struct to return undefined because the sentinel header is still stored in `unusedEmHeader`. This would prevent further reading of the message at all essentially.

Reply:
I updated the comment to hopefully be more clear